applications/CompressiblePotentialFlowApplication/custom_elements/adjoint_finite_difference_potential_flow_element.cpp

```diff
@@ -0,0 +1,126 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:		 BSD License
+//					 Kratos default license: kratos/license.txt
+//
+//
+//  Main authors:    Marc Nuñez, based on A. Geiser, M. Fusseder, I. Lopez and R. Rossi work
+//
+#include "compressible_potential_flow_application_variables.h"
+#include "incompressible_potential_flow_element.h"
+#include "compressible_potential_flow_element.h"
+#include "adjoint_finite_difference_potential_flow_element.h"
+
+namespace Kratos
+{
+    template <class TPrimalElement>
+    Element::Pointer AdjointFiniteDifferencePotentialFlowElement<TPrimalElement>::Create(IndexType NewId, NodesArrayType const& ThisNodes, typename PropertiesType::Pointer pProperties) const
+    {
+        KRATOS_TRY
+          return Kratos::make_intrusive<AdjointFiniteDifferencePotentialFlowElement<TPrimalElement>>(NewId, this->GetGeometry().Create(ThisNodes), pProperties);
+        KRATOS_CATCH("");
+    }
+
+    template <class TPrimalElement>
+    Element::Pointer AdjointFiniteDifferencePotentialFlowElement<TPrimalElement>::Create(IndexType NewId, typename GeometryType::Pointer pGeom, typename PropertiesType::Pointer pProperties) const
+    {
+        KRATOS_TRY
+            return Kratos::make_intrusive<AdjointFiniteDifferencePotentialFlowElement<TPrimalElement>>(NewId, pGeom, pProperties);
+        KRATOS_CATCH("");
+    }
+
+    template <class TPrimalElement>
+    Element::Pointer AdjointFiniteDifferencePotentialFlowElement<TPrimalElement>::Clone(IndexType NewId, NodesArrayType const& ThisNodes) const
+    {
+        KRATOS_TRY
+        return Element::Pointer(new AdjointFiniteDifferencePotentialFlowElement(NewId, this->GetGeometry().Create(ThisNodes), this->pGetProperties()));
+        KRATOS_CATCH("");
+    }
+
+
+    template <class TPrimalElement>
+    void AdjointFiniteDifferencePotentialFlowElement<TPrimalElement>::CalculateSensitivityMatrix(const Variable<array_1d<double,3> >& rDesignVariable,
+                                            Matrix& rOutput,
+                                            const ProcessInfo& rCurrentProcessInfo)
+    {
+        KRATOS_TRY;
+        const double delta = this->GetPerturbationSize();
```

Comment:
I think we discussed this in another PR, but I think that we are mixing the finite difference from the element and the response function.  Would it be better to pass the PerturbationSize within the rCurrentProcessInfo? And then do something like:

```suggestion
        const double delta = rCurrentProcessInfo[SCALE_FACTOR];
```

Also, maybe a better naming for the variable might be a good idea. In the structural mechanics application they do:

```cpp
const double perturbation_size = rCurrentProcessInfo[PERTURBATION_SIZE];
```

This would be anyway for another PR.

Reply:
You are right! I would leave this for another PR since I also wanted to add the possibility to define the perturbation size in terms of the nodal size